infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftProjectFactory.java

```diff
@@ -84,67 +76,38 @@ public OpenShiftProjectFactory(
     this.clientFactory = clientFactory;
   }
 
-  /**
-   * Creates a OpenShift project for the specified workspace.
-   *
-   * <p>The project name will be chosen according to a configuration, and it will be prepared
-   * (created if necessary).
-   *
-   * @param workspaceId identifier of the workspace
-   * @return created project
-   * @throws InfrastructureException if any exception occurs during project preparing
-   */
-  public OpenShiftProject create(String workspaceId) throws InfrastructureException {
-    final String projectName;
-    try {
-      projectName = evalNamespaceName(workspaceId, EnvironmentContext.getCurrent().getSubject());
-    } catch (NotFoundException | ServerException | ConflictException | ValidationException e) {
-      throw new InfrastructureException(
-          format(
-              "Failed to evaluate the project name to use for workspace %s. The error message"
-                  + " was: %s",
-              workspaceId, e.getMessage()),
-          e);
-    }
-    OpenShiftProject osProject = doCreateProject(workspaceId, projectName);
-    osProject.prepare();
+  public OpenShiftProject getOrCreate(RuntimeTarget target) throws InfrastructureException {
```

Comment:
same as k8s

Reply:
It isn't. (just trying to win the contest for the shortest PR review comment)